controls/V-72845.rb

```diff
@@ -29,55 +17,84 @@
+  category of the information system or the criticality of the update (i.e.,
+  severity of the vulnerability related to the discovered flaw).
+
+  This requirement will apply to software patch management solutions that are
+  used to install patches across the enclave and also to applications themselves
+  that are not part of that patch management solution. For example, many browsers
+  today provide the capability to install their own patch software. Patch
+  criticality, as well as system criticality, will vary. Therefore, the tactical
+  situations regarding the patch management process will also vary. This means
+  that the time period utilized must be a configurable parameter. Time frames for
+  application of security-relevant software updates may be dependent upon the
+  Information Assurance Vulnerability Management (IAVM) process.
+
+  The application will be configured to check for and install
+  security-relevant software updates within an identified time period from the
+  availability of the update. The specific time period will be defined by an
+  authoritative source (e.g., IAVM, CTOs, DTMs, and STIGs)."
+
   impact 0.7
   tag "severity": "high"
   tag "gtitle": "SRG-APP-000456-DB-000390"
   tag "gid": "V-72845"
   tag "rid": "SV-87497r1_rule"
   tag "stig_id": "PGS9-00-000300"
+  tag "fix_id": "F-79287r1_fix"
   tag "cci": ["CCI-002605"]
   tag "nist": ["SI-2 c", "Rev_4"]
-
-  tag "check": "If new packages are available for PostgreSQL, they can be
+  tag "false_negatives": nil
+  tag "false_positives": nil
+  tag "documentable": false
+  tag "mitigations": nil
+  tag "severity_override_guidance": false
+  tag "potential_impacts": nil
+  tag "third_party_tools": nil
+  tag "mitigation_controls": nil
+  tag "responsibility": nil
+  tag "ia_controls": nil
+  desc "check", "If new packages are available for PostgreSQL, they can be
   reviewed in the package manager appropriate for the server operating system:
+
   To list the version of installed PostgreSQL using psql:
+
   $ sudo su - postgres
   $ psql -–version
+
   To list the current version of software for RPM:
+
   $ rpm -qa | grep postgres
+
   To list the current version of software for APT:
+
   $ apt-cache policy postgres
+
   All versions of PostgreSQL will be listed on:
+
   http://www.postgresql.org/support/versioning/
+
   All security-relevant software updates for PostgreSQL will be listed on:
+
   http://www.postgresql.org/support/security/
+
   If PostgreSQL is not at the latest version, this is a finding.
+
   If PostgreSQL is not at the latest version and the evaluated version has CVEs
   (IAVAs), then this is a CAT I finding."
 
-  tag "fix": "Institute and adhere to policies and procedures to ensure that
+  desc "fix", "Institute and adhere to policies and procedures to ensure that
   patches are consistently applied to PostgreSQL within the time allowed."
 
-  only_if { false }
+  sql = postgres_session(pg_dba, pg_dba_password, pg_host)
```

Comment:
Let's make this implementation similar to V-72917 and V-73011. Make sure the versions listed with 'rpm -qa | grep postgres' are not different than the currently installed version. 

Do this for both RPM and APT.